Specifying `dockerFile` for `DockerBuildImage` fails

<!--- Please follow the instructions below. We receive dozens of issues every week, -->
<!--- so to stay productive, we will close issues that don't provide enough information. -->

<!--- Provide a brief summary of the issue in the title above -->

### Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
The task should execute successfully.

### Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->
The task fails with the following call stack:
```
Caused by: groovy.lang.MissingMethodException: No signature of method: com.github.dockerjava.core.command.BuildImageCmdImpl.withDockerfile() is applicable for argument types: (org.gradle.api.internal.file.DefaultFilePropertyFactory$FixedFile) values: [/Users/daniel/boogie/boogie-platform/.docker/application/Dockerfile]
Possible solutions: withDockerfile(java.io.File), withDockerfilePath(java.lang.String)
        at com.bmuschko.gradle.docker.tasks.image.DockerBuildImage.runRemoteCommand(DockerBuildImage.groovy:144)
        at com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask$_start_closure1.doCall(AbstractDockerRemoteApiTask.groovy:63)
        at com.bmuschko.gradle.docker.utils.DockerThreadContextClassLoader.withContext(DockerThreadContextClassLoader.groovy:61)
        at com.bmuschko.gradle.docker.utils.DockerThreadContextClassLoader$withContext.call(Unknown Source)
        at com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask.runInDockerClassPath(AbstractDockerRemoteApiTask.groovy:110)
        at com.bmuschko.gradle.docker.tasks.AbstractDockerRemoteApiTask.start(AbstractDockerRemoteApiTask.groovy:62)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:48)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:41)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:704)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:671)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:117)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:300)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:292)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:174)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:90)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:106)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:85)
        ... 37 more
```

### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

### Steps to Reproduce (for bugs)
<!--- Provide a self-contained example project (as an attached archive or a Github project). -->
<!--- In the rare cases where this is infeasible, we will also accept a detailed set of instructions. -->

### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

--- a/src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy
+++ b/src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy
@@ -43,6 +43,22 @@ class DockerBuildImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
         result.output.contains("Created image with ID")
     }
 
+    def "can build image in a parent context"() {
+        buildFile << imageCreation()
+        buildFile << """
+            buildImage {
+                inputDir = projectDir
+                dockerFile = dockerfile.destFile
+            }
+        """
+
+        when:
+        BuildResult result = build('buildImage')
+
+        then:
+        result.output.contains("Created image with ID")
+    }
+
     @Unroll
     def "can build image with labels"(String gradleTaskDefinition) {
         buildFile << gradleTaskDefinition

--- a/src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy
+++ b/src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy
@@ -138,7 +138,7 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
             logger.quiet "Using Dockerfile '${dockerFile.get()}'"
             buildImageCmd = dockerClient.buildImageCmd()
                     .withBaseDirectory(inputDir.get().asFile)
-                    .withDockerfile(dockerFile.get())
+                    .withDockerfile(dockerFile.get().asFile)
         } else {
             buildImageCmd = dockerClient.buildImageCmd(inputDir.get().asFile)
         }

